src/rdkafka_broker.c

```diff
@@ -3830,14 +3852,32 @@ static int rd_kafka_broker_fetch_toppars (rd_kafka_broker_t *rkb, rd_ts_t now) {
 
         if (unlikely(rkb->rkb_active_toppar_cnt == 0))
                 return 0;
-
+#if WITH_ZSTD
```

Comment:
don't need a conditional here, just go with the version with the maximum overhead.